packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/zoomlevel/ZoomLevelFeature.java

```diff
@@ -28,18 +29,24 @@ public ZoomLevelFeature(CameraProperties cameraProperties) {
     sensorArraySize = cameraProperties.getSensorInfoActiveArraySize();
 
     if (sensorArraySize == null) {
-      maximumZoomLevel = MINIMUM_ZOOM_LEVEL;
+      maximumZoomLevel = minimumZoomLevel;
       hasSupport = false;
       return;
     }
 
-    Float maxDigitalZoom = cameraProperties.getScalerAvailableMaxDigitalZoom();
-    maximumZoomLevel =
-        ((maxDigitalZoom == null) || (maxDigitalZoom < MINIMUM_ZOOM_LEVEL))
-            ? MINIMUM_ZOOM_LEVEL
-            : maxDigitalZoom;
+    if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
+      minimumZoomLevel = cameraProperties.getScalerMinZoomRatio();
+      maximumZoomLevel = cameraProperties.getScalerMaxZoomRatio();
```

Comment:
Can you leave a comment somewhere around here as to why this is done for API 30+?